### PR TITLE
Backport #1326 to maint-1.2

### PR DESCRIPTION
**Pull request by jan-cerny**
This pull request adds SCAP 1.3 source datastream XML  schema. OASIS XML Catalog schema and WC3 XLink schema, that are shared  between SCAP 1.3 and 1.2 are moved to /schemas/common directory to avoid duplication. So far version 1.2 was hard-coded, therefore I have added a function that detects the version from the datastream using data-stream-collection@schematron-version attribute. Some of the 1.2 datastreams in our test suite have set the version to 1.0, so I fixed this.

This has been previously merged into maint-1.3 as #1326. However, users of maint-1.2 releases might also like to also validate SCAP 1.3 content. The tests had to be changed to use Automake.

### CODE REVIEW

**Comment by jan-cerny** [comment]
I'm now facing a problem only on RHEL6 for `tests/DS/validate/all.sh` (valid-1.3-sds) and  `tests/DS/all.sh` (test_ds_1_3_continue_without_remote_resources).

The messages in the test log are very similar in both cases.
```
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1446: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1459: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1472: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1485: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1652: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1665: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1678: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
File '/root/openscap/schemas/oval/5.11.2/oval-definitions-schema.xsd' line 1691: local union type: A type, derived by list or union, must have the simple ur-type definition as base type, not '{http://oval.mitre.org/XMLSchema/oval-definitions-5}(NULL)'.
OpenSCAP Error: Could not parse XML schema [validate.c:113]
Invalid SCAP Source Datastream (1.3) content in ./sds-1.3-valid.xml. [oscap_source.c:343]
```
Notice that both `sds-1.3-valid.xml` and `remote_content_1.3.ds.xml` contain OVAL 5.11 document, not 5.11.2 document, but that's because the datastream schema imports explicit version of OVAL schema, which means the actually used version of datastream components is ignored and they're always validated against version specified in the datastream schema. That's not a new thing, we had to do a change of the SCAP 1.2 datastream schema in https://github.com/OpenSCAP/openscap/pull/58

Based on the error message I have found that we have forgotten to edit the OVAL 5.11.2 schemas as we already did in  9be08e3f78b00a93477c19767bb3f06f40d169a2, d4b8cc8039fb6dd6f1e7f9f331244c04c3574ea6,  8ba623120fc9f479285f9d6032cb925db420011d, 8ba623120fc9f479285f9d6032cb925db420011d and other commits. I suggest doing it in a separate PR because it's unrelated to this PR/

**Comment by jan-cerny** [comment]
Blocked by #1336

**Comment by jan-cerny** [comment]
After #1336 has been merged it's unblocked. I have rebased it on the top of `maint-1.2`.

File: tests/DS/sds_detect_version/test_detect_version.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# OpenSCAP Test Suite
+#
+# Authors:
+#      Jan Černý <jcerny@redhat.com>
+
+. $builddir/tests/test_common.sh
+
+set -e -o pipefail
+
+function test_oscap_info {
```
Comment (ggbecker) [inline tests/DS/sds_detect_version/test_detect_version.sh]: What about change the test to detect which sds are available in the project and test all of them? The Datastream can easily be created by copying and changing the version.
```suggestion
function test_oscap_info {
	version="$1"
	stdout="$(mktemp)"
	stderr="$(mktemp)"
	ds="$(mktemp)"
    cp scap-ds.xml $ds
    sed -i "s/X.X/${version}/g" $ds

	$OSCAP info $ds > $stdout 2> $stderr
	[ ! -s $stderr ]
	grep -q "Version: $version" $stdout
	rm $stdout
	rm $stderr
    rm $ds
}

SDS=$(find ../../../schemas/sds -maxdepth 1 -mindepth 1 -type d -printf '%f\n')
echo $SDS

for sds_version in $SDS
do
    test_oscap_info $sds_version
done
```

File: tests/DS/sds_detect_version/scap-1.3-ds.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <ns0:checks>
+      <ns0:component-ref id="scap_org.open-scap_cref_ssg-rhel8-oval.xml" ns1:href="#scap_org.open-scap_comp_ssg-rhel8-oval.xml"/>
+    </ns0:checks>
+  </ns0:data-stream>
+  <ns0:component id="scap_org.open-scap_comp_ssg-rhel8-oval.xml" timestamp="2019-05-10T08:18:44">
+    <ns4:oval_definitions xsi:schemaLocation="http://oval.mitre.org/XMLSchema/oval-common-5 oval-common-schema.xsd         http://oval.mitre.org/XMLSchema/oval-definitions-5 oval-definitions-schema.xsd         http://oval.mitre.org/XMLSchema/oval-definitions-5#independent independent-definitions-schema.xsd         http://oval.mitre.org/XMLSchema/oval-definitions-5#unix unix-definitions-schema.xsd         http://oval.mitre.org/XMLSchema/oval-definitions-5#linux linux-definitions-schema.xsd">
+      <ns4:generator>
+        <ns6:product_name>combine_ovals.py from SCAP Security Guide</ns6:product_name>
+        <ns6:product_version>ssg: [0, 1, 44], python: 3.7.3</ns6:product_version>
+        <ns6:schema_version>5.11</ns6:schema_version>
```
Comment (ggbecker) [inline tests/DS/sds_detect_version/scap-1.3-ds.xml]: It doesn't affect the test, but shouldn't this be 5.11.2?

Comment (jan-cerny) [reply tests/DS/sds_detect_version/scap-1.3-ds.xml]: @ggbecker great catch! I'll fix that.

File: tests/DS/sds_detect_version/test_detect_version.sh (hunk 1 above)
Comment (jan-cerny) [reply tests/DS/sds_detect_version/test_detect_version.sh]: @ggbecker Great idea! Please submit that as a separate PR.

**Comment by jan-cerny** [comment]
@ggbecker I have bumped the OVAL schema version

File: tests/DS/sds_detect_version/test_detect_version.sh (hunk 1 above)
Comment (ggbecker) [reply tests/DS/sds_detect_version/test_detect_version.sh]: Sure, I'll create a new PR to propose this solution. Thanks

**Comment by ggbecker** [comment]
LGTM

**Pull request merged**